### DISCUSSION TITLE
Playing audio from another tab feeds into BirdNET with disastrous results

### DISCUSSION THREAD

**Discussion by ThickKnee** [mcguirepr89/BirdNET-Pi#477] [General]
If I play an audio in a tab other than the BirdNET tab, it feeds into BirdNET. I can hear it in the Live Audio.

BirdNET analyses this audio and sometimes produces an ID. It certainly will produce an ID if I play an .mp3 file from the Birdnet folders. It also happens if I listen to Internet radio.

If I stop the audio I still hear it in the Live Audio and it seems to repeat endlessly. This leads to a very large number of false IDs being recorded in BirdNET's database.

This is not just an inconvenience as we like to listen to calls from e-bay or xeno to validate BirdNET's IDs.

**Comment by mcguirepr89** [comment]
This certainly makes no sense at all -- anyone have any ideas what the hell??

**Reply by ThickKnee** [reply]
Hi Patrick.

I wondered how this problem could be tolerated -- so it's only me?

It seems to be a permanent problem, not intermittent. It means we can easily contaminate out BirdNET data.

John.

**Comment by jds3d** [comment]
are you playing the audio within short distance of the birdnetpi microphone?

**Reply by ThickKnee** [reply]
Good thought, but no. We have actually done that -- two PCs on the one desk; one running BirdNET, the other playing calls from e-bird. Naturally BirdNET hears & processes the e-bird output so it will produce a perfect ID (embarrassing if it didn't). So that's a situation to be aware of.

But this is different. It happened tonight while I had NO microphone connected to the Pi4 and I was listening with earplugs -- no speaker. It also repeats endlessly after shutting down the audio. I'm satisfied it's happening internally, i.e., from one program directly to the other.

Is the audio jack on the Rpi4 purely an output device? Can it be used as an audio input? Just a thought. It doesn't appear in the choices of audio inputs in the audio control menus.

John.

**Reply by CaiusX** [reply]
Hi @ThickKnee 

Are you running any other audio software on the PI which could be "mixing" other sound sources into the sound layers (ffmpeg/alsa/pulse) - Jack comes to mind...

Dennis

**Reply by ehpersonal38** [reply]
> It happened tonight while I had NO microphone connected to the Pi4 and I was listening with earplugs

So you unplugged the microphone from the Pi, and there was still a live audio feed playing with sound on the BirdNET-Pi interface?

**Reply by ThickKnee** [reply]
_Are you running any other audio software on the PI_ 

Not that I'm aware of. I haven't played around with any such software while using this BirdNET image. I don't know Jack. 
I have in earlier sessions sent the audio output to Bluetooth speakers.

_So you unplugged the microphone from the Pi,_

No -- I didn't have a mic plugged in at any time.

**Reply by ThickKnee** [reply]
> So you unplugged the microphone from the Pi, and there was still a live audio feed playing with sound on the BirdNET-Pi interface?

Actually that does occur and I thought it was normal as I've read comments from others about the lag between recording and the Live Audio. But I haven't seen that as a problem.

**Reply by ehpersonal38** [reply]
Could you try changing the audio card in Tools -> Settings -> Advanced Settings from "default" to your microphone? 
This command will list the available audio cards, one will be your mic: `aplay -L | awk -F, '/^hw:/ { print $1 }'`

Then restart core services and try again

**Reply by ThickKnee** [reply]
Okay, I'll try that later today and come back to you. Thanks.

**Comment by ThickKnee** [comment]
This screenshot is some evidence of the problem as it occurred tonight. If the text is legible you can see 40+ IDs of the Superb Lyrebird as a result of me playing its .mp3 file from a Birdnet folder. Also the large number of ID's of kiwis. (We don't have kiwis in Oz.)
(edit: my guess is the kiwis resulted from playing the Internet radio.)

![2022-08-02-233702_1600x900_scrot](https://user-images.githubusercontent.com/110239701/182388548-da1f2e1e-b010-4c7a-9336-5a29dc317348.png)

**Comment by DD4WH** [comment]
If I interpret your screenshot correctly, you are playing the audio file on the same Raspberry Pi that is running BirdNET-Pi ? Is that true ?
If so, I could imagine why this confuses the BirdNET-Pi . . . 
And this also explains how your BirdNET-Pi can listen to birds when no mic is connected as you stated in one of your messages.
Strange  . . .
The solution could be:
* use a different PC, RPi or whatever to listen to Internet Radio :-)

**Comment by CaiusX** [comment]
Hi @ThickKnee 

WHAT A FANTASTIC FIND!!  

I can reproduce your experience: 
With no microphone plugged in/recognized, BirdNET-Pi defaults to another sound source - whatever Pulse thinks is the "default" source is- which can be a streaming service (xeno-canto, youtube livestream, your machine playing VLC, audacity, an AM country station or nordic goth).

Soooooo- to answer your question in https://github.com/mcguirepr89/BirdNET-Pi/discussions/473#discussion-4262546, simply unplug the microphone and play the MP3, which will give date and time-stamped recognitions.

Now the question is how to distinguish what was from the microphone and what was from a played sample?

Suggestions please, a couple from me:
- each detected sample should have a "Source" field in the database - whatever  the arecord/Pulse sound source is at the time of recording
- each sound source should have a separate database and sample storage directory?
- ???

I'll look at implementing in Python, but may be better to do in the system @mcguirepr89 ?

Best
AP

**Reply by nilspupils** [reply]
Great idea to turn this into a feature!! I wonder where the cycling comes from though, probably related to a buffer which is read again and again?

**Reply by ThickKnee** [reply]
Yay! It's not just me.

So my problem is a pointer to how my need to feed recorded calls into BirdNET-pi can be met. This would be fabulous. We've just spent seven days in a bush location using BirdNET every day. Identified new species in a place we thought we were familiar with. It's a great tool but must be used "under supervision".

I believe capturing bird calls on a recorder and feeding into BirdNET at a later time is more useful than feeding an RPi directly from  a live mic. The first allows placement of the recorder overnight in a remote location, while the second has to be done from our dining room table.

Recording the audio source would be helpful but that issue exists now. Ideally, you want to record location and time of the actual calls. All that is recorded now is time of analysis which is not time of recording (although close if done live, but simply wrong if done from a recording).

So glad my problem has lead to a potential feature.
